lib/schema.js

```diff
@@ -81,14 +81,15 @@ Schema.prototype.create = function(obj, schema){
 			}
 
 			if (propertySchema.type === 'Array') {
-				return Promise.each(value, function(v, i) {
-					this.i = i;
-					return propertySchema.of.create({subSchema: v});
+				return Promise.map(value, function(v, i) {
```

Comment:
This is the standard ```.map()``` arguments. The predicate functions starts with two arguments. The first one is the current iteration value and the second the index. 
I'll change their names if you do not understand them.